src/Templates/BlockTemplatesCompatibility.php

```diff
@@ -361,4 +361,134 @@ protected function inject_attribute( &$block ) {
 		}
 	}
 
+	/**
+	 * For compatibility reason, we need to wrap the Single Product template in a div with specific class.
+	 * For more details, see https://github.com/woocommerce/woocommerce-blocks/issues/8314.
+	 *
+	 * @param string $template_content Template Content.
+	 * @return string Wrapped template content inside a div.
+	 */
+	public static function wrap_single_product_template( $template_content ) {
+		$parsed_blocks  = parse_blocks( $template_content );
+		$grouped_blocks = self::group_blocks( $parsed_blocks );
+
+		$single_product_template_blocks = array( 'woocommerce/breadcrumbs', 'woocommerce/product-gallery-image', 'woocommerce/product-add-to-cart', 'woocommerce/product-details', 'woocommerce/add-to-cart-form' );
```

Comment:
When using the classic template, the breadcrumbs are shown outside of the `.product` div, so I think we should exclude `breadcrumbs` from this list.

![imatge](https://user-images.githubusercontent.com/3616980/218486401-ed485072-3f9e-456f-9a69-9e1ed6cc444f.png)

Also `woocommerce/product-add-to-cart` is used in the Products block, but not in the Single Product template, no? Should we remove it from the list as well? :thinking: